rocketpool/watchtower/submit-rewards-tree-rolling.go

```diff
@@ -391,39 +389,35 @@ func (t *submitRewardsTree_Rolling) getTrueRewardsIntervalSubmissionSlot(targetS
 }
 
 // Checks to see if an existing rewards file is still valid and whether or not it should be regenerated or just resubmitted
-func (t *submitRewardsTree_Rolling) isExistingFileValid(rewardsTreePath string, intervalsPassed uint64, nodeAddress common.Address, isInOdao bool) (rprewards.IRewardsFile, []byte, bool, bool) {
+func (t *submitRewardsTree_Rolling) isExistingFileValid(rewardIndex uint64, intervalsPassed uint64, nodeAddress common.Address, isInOdao bool) (*rprewards.LocalRewardsFile, bool, bool) {
+	rewardsTreePath := t.cfg.Smartnode.GetRewardsTreePath(rewardIndex, true)
+
 	// Check if the rewards file exists
 	_, err := os.Stat(rewardsTreePath)
 	if os.IsNotExist(err) {
-		return nil, nil, false, true
+		return nil, false, true
 	}
 	if err != nil {
 		t.log.Printlnf("%s WARNING: failed to check if [%s] exists: %s; regenerating file...\n", t.logPrefix, rewardsTreePath, err.Error())
-		return nil, nil, false, true
+		return nil, false, true
 	}
 
 	// The file already exists, attempt to read it
-	filename := filepath.Base(rewardsTreePath)
-	fileBytes, err := os.ReadFile(rewardsTreePath)
+	localRewardsFile, err := rprewards.ReadLocalRewardsFile(rewardsTreePath)
 	if err != nil {
 		t.log.Printlnf("%s WARNING: failed to read %s: %s; regenerating file...\n", t.logPrefix, rewardsTreePath, err.Error())
-		return nil, nil, false, true
+		return nil, false, true
 	}
 
-	// Unmarshal it
-	proofWrapper, err := rprewards.DeserializeRewardsFile(fileBytes)
-	if err != nil {
-		t.log.Printlnf("%s WARNING: failed to deserialize %s: %s; regenerating file...\n", t.logPrefix, rewardsTreePath, err.Error())
-		return nil, nil, false, true
-	}
+	proofWrapper := localRewardsFile.Repr()
 	header := proofWrapper.GetHeader()
 
 	if isInOdao {
 		// Get the CID for it
-		cid, err := rprewards.GetCidForRewardsFile(proofWrapper, filename)
```

Comment:
this line is the buggy one.

`proofWrapper` is decompressed, `GetCidForRewardsFile` ultimately calls `SingleFileDirIPFSCid` which does compress it eventually, but doesn't add .zst to filename before calculating the cid:
https://github.com/rocket-pool/smartnode/pull/432/files#diff-51c4c70c13d682999d12284148b72676b8fe2672757f756401d049f950c18793L83

because `filename` here is the uncompressed file name, the cid would be incorrect.